test/unittest/tests/test_bounds_restore.cpp

```diff
@@ -0,0 +1,63 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+
+#include "utils.h"
+#include "wiredtiger.h"
+#include "wrappers/connection_wrapper.h"
+#include "wt_internal.h"
+
+int
+validate_cursor_bounds_restore(WT_CURSOR *cursor, uint64_t original_cursor_flags)
+{
+    assert(cursor->flags == original_cursor_flags);
+    return (0);
+}
+
+TEST_CASE("Bounds save and restore flag logic", "[bounds_restore]")
+{
+    uint64_t original_cursor_flags;
+    WT_CURSOR mock_cursor;
+    WT_CURSOR_BOUNDS_STATE mock_state;
+
+    ConnectionWrapper conn(DB_HOME);
+    WT_SESSION_IMPL *session = conn.createSession();
+
+    mock_state.lower_bound = NULL;
+    mock_state.upper_bound = NULL;
+
+    mock_cursor.lower_bound.size = 0;
+    mock_cursor.lower_bound.data = NULL;
+    mock_cursor.upper_bound.size = 0;
+    mock_cursor.upper_bound.data = NULL;
+
+    // Save bounds flags and ensure that the restore logic correctly restores the desired flags.
+    SECTION("Save non-empty non-inclusive bounds flags and restore")
+    {
+        F_SET(&mock_cursor, WT_CURSTD_BOUND_UPPER);
+        F_SET(&mock_cursor, WT_CURSTD_BOUND_LOWER);
+        original_cursor_flags = mock_cursor.flags;
+        REQUIRE(__wt_cursor_bounds_save(session, &mock_cursor, &mock_state) == 0);
+        REQUIRE(__wt_cursor_bounds_restore(session, &mock_cursor, &mock_state) == 0);
+        REQUIRE(validate_cursor_bounds_restore(&mock_cursor, original_cursor_flags) == 0);
+        __wt_scr_free(session, &mock_state.lower_bound);
+        __wt_scr_free(session, &mock_state.upper_bound);
+    }
+
+    mock_cursor.flags = 0;
```

Comment:
I'd suggest moving the initialisation of mock_cursor.flags to line 35 or 39 so that's it's with the other initialisations. The code outside a SECTION is run separately for each SECTION (in this case twice, as there are two SECTIONs), so there's no need to reset for the next SECTION.

Reply:
Good point, thanks! [0cb5d18](https://github.com/wiredtiger/wiredtiger/pull/8335/commits/0cb5d1872bce15272b3e941cf856b34d45d8d75c)